tools/CouchDB_Confirm_Integrity.php

```diff
@@ -94,7 +94,12 @@ function run()
                 ]
             );
 
-            if (!empty($sqlDB) && $sqlDB['cActive'] == 'N') {
+            if (empty($sqlDB)) {
+                print "PSCID $pscid VL $vl does not exist but $row[id] still exists.
+                Deleting Doc. \n";
+
+                $this->CouchDB->deleteDoc($row['id']);
```

Comment:
If you change the query on line 87 to 
```
SELECT c.*, s.Visit_label, s.Active, c.Active as cActive
                FROM candidate c
                LEFT JOIN session s USING (CandID)
                WHERE c.Active='Y' AND c.PSCID=:PID AND s.Visit_label=:VL
```

Then you can keep these lines and remove lines 102 to 106. you are basically doing a PHP check here that can be done in SQL. something similar might be feasable for session... 
just to remove useless code. no change in behaviour

Reply:
I don't think that would work because it would break the check on line 102.

Reply:
... read my comment, I'm saying remove line 102 to 106

Reply:
That seems like it would make the diagnostics/logging less explicit. Is there any benefit other than a couple fewer lines of code?

Reply:
none, just code cleanliness